engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala

```diff
@@ -3794,6 +3794,268 @@ class RuntimeVisualizationsTest extends AnyFlatSpec with Matchers {
+      context.executionComplete(contextId)
+    )
+    val Some(data) = attachVisualizationResponses.collectFirst {
+      case Api.Response(
+            None,
+            Api.VisualizationUpdate(
+              Api.VisualizationContext(
+                `visualizationId`,
+                `contextId`,
+                `idRes`
+              ),
+              data
+            )
+          ) =>
+        data
+    }
+    new String(data, StandardCharsets.UTF_8) shouldEqual "(Mk_Newtype 42)"
+  }
+
+  it should "visualize autoscoped atomconstructor" in withContext() { context =>
+    val contextId       = UUID.randomUUID()
+    val requestId       = UUID.randomUUID()
+    val visualizationId = UUID.randomUUID()
+    val moduleName      = "Enso_Test.Test.Main"
+    val metadata        = new Metadata
+
+    val idX    = metadata.addItem(103, 29, "aaaa")
+    val idCons = metadata.addItem(116, 15, "cccc")
+    val idRes  = metadata.addItem(137, 1, "eeee")
+
+    val code =
+      """from Standard.Base import all
+        |
+        |type Newtype
+        |    Mk_Newtype value
+        |
+        |    fix t:Newtype = t
+        |main =
+        |    x = Newtype.fix (..Mk_Newtype 42)
+        |    x
+        |""".stripMargin.linesIterator.mkString("\n")
+    val contents = metadata.appendToCode(code)
+    val mainFile = context.writeMain(contents)
+
+    metadata.assertInCode(idX, code, "Newtype.fix (..Mk_Newtype 42)")
+    metadata.assertInCode(idCons, code, "..Mk_Newtype 42")
+    metadata.assertInCode(idRes, code, "x")
+
+    // create context
+    context.send(Api.Request(requestId, Api.CreateContextRequest(contextId)))
+    context.receive shouldEqual Some(
+      Api.Response(requestId, Api.CreateContextResponse(contextId))
+    )
+
+    // Open the new file
+    context.send(
+      Api.Request(requestId, Api.OpenFileRequest(mainFile, contents))
+    )
+    context.receive shouldEqual Some(
+      Api.Response(Some(requestId), Api.OpenFileResponse)
+    )
+
+    // push main
+    val item1 = Api.StackItem.ExplicitCall(
+      Api.MethodPointer(moduleName, "Enso_Test.Test.Main", "main"),
+      None,
+      Vector()
+    )
+    context.send(
+      Api.Request(requestId, Api.PushContextRequest(contextId, item1))
+    )
+    context.receiveNIgnorePendingExpressionUpdates(
+      5
+    ) should contain theSameElementsAs Seq(
+      Api.Response(requestId, Api.PushContextResponse(contextId)),
+      TestMessages.update(
+        contextId,
+        idCons,
+        ConstantsGen.FUNCTION,
+        methodCall = None,
+        payload    = Api.ExpressionUpdate.Payload.Value(None)
+      ),
+      TestMessages.update(
+        contextId,
+        idX,
+        s"$moduleName.Newtype",
+        methodCall = Some(
+          Api.MethodCall(
+            Api
+              .MethodPointer(moduleName, s"$moduleName.Newtype", "fix")
+          )
+        ),
+        payload = Api.ExpressionUpdate.Payload.Value(None)
```

Comment:
A call to `Some(MethodCall(MethodPointer("Enso_Test.Test.Main", "Enso_Test.Test.Main.Newtype", "Mk_Newtype")` is missing somewhere here.